test/unit/argument_parser/format_parse_test.cpp

```diff
@@ -641,6 +641,26 @@ TEST(parse_type_test, parse_error_enum_option)
     EXPECT_THROW(parser.parse(), seqan3::argument_parser_error);
 }
 
+TEST(parse_test, issue2464_enum_user_input_error)
+{
+    // see issue https://github.com/seqan/seqan3/issues/2464
+    Other::bar option_value{};
+
+    {
+        const char * argv[] = {"./argument_parser_test", "-m", "9"};
+        seqan3::argument_parser parser{"test_parser", 3, argv, seqan3::update_notifications::off};
+        parser.add_option(option_value, 'e', "enum-option", "this is an enum option.");
+        EXPECT_THROW(parser.parse(), seqan3::argument_parser_error);
+    }
+    {
+        Other::bar option_value{};
+        const char * argv[] = {"./argument_parser_test", "-m", "nine"};
+        seqan3::argument_parser parser{"test_parser", 3, argv, seqan3::update_notifications::off};
+        parser.add_option(option_value, 'e', "enum-option", "this is an enum option.");
+        EXPECT_THROW(parser.parse(), seqan3::argument_parser_error);
+    }
```

Comment:
This seems like the same test as before?

Maybe you could test the variant where you parse a vector of enums and then one of the enums has an invalid value.

Reply:
Good idea!

Is there a possibility to test for the Error message aswell. I found no tests about the messages at all.

Reply:
I don't think there is functionality for that.
You could use a try-catch block, and check the string equality of `error.what()` and the expected error message (which we then should add as a macro).

But I think we only ever check for error types.